boards/xtensa/esp32/common/src/esp32_tca9548a.c

```diff
@@ -0,0 +1,145 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <debug.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/i2c/i2c_master.h>
+#include <nuttx/i2c/tca9548a.h>
+
+#include "esp32_board_i2c.h"
+#include "esp32_i2c.h"
+#include "esp32_tca9548a.h"
+
+FAR struct tca9548a_dev_s *g_tca9448a_devs[8] =
+                                                {
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL
+                                                };
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_tca9548a_initialize
+ *
+ * Description:
+ *   Initialize and register the TCA9548A Multiplexer.
+ *
+ * Input Parameters:
+ *   devno - The device number associated to I2C address, 0=0x70 ... 7=0x77
+ *   busno - The I2C bus number
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int board_tca9548a_initialize(int devno, int busno)
+{
+  struct i2c_master_s *i2c;
+  uint8_t addr;
+
+  i2cinfo("Initializing TCA9548A!\n");
+
+  /* TCA9548A valid addresses are 0:0x70 - 7:0x77 */
+
+  if (devno < TCA9548A_SEL_CH0 && devno > TCA9548A_SEL_CH7)
+    {
+      i2cerr("Invalid devno: %d! Valid range: 0 to 7!\n", devno);
+      return -EINVAL;
+    }
+
+  addr = TCA9548A_BASEADDR0 + devno;
```

Comment:
This part and part in `esp32_i2cmux_getmaster` are really confusing to me. All the time compare with `TCA9548A_SEL_CH`.
I would suggest to rework this into:
```suggestion
  if (devno < 0 || devno > (TCA9548A_BASEADDR7 - TCA9548A_BASEADDR0))
    {
      i2cerr("Invalid devno: %d! Valid range: 0 to 7!\n", devno);
      return -EINVAL;
    }

  addr = TCA9548A_BASEADDR0 + devno;
```
And the same in `esp32_i2cmux_getmaster`

or
```
#define TCA9548A_DEV_CNT (TCA9548A_BASEADDR7 - TCA9548A_BASEADDR0 + 1)
```
and use it in `FAR struct tca9548a_dev_s *g_tca9448a_devs[TCA9548A_DEV_CNT]` and here in check condition `if (devno < 0 || devno >= TCA9548A_DEV_CNT)`.

Anyway even with current logic need a change from `&&` to `||`

Reply:
You are right, comparing it to TCA9548A_SEL_CH7 at "board_tca9548a_initialize" is generating confusion:2
```
if (devno < TCA9548A_SEL_CH0 && devno > TCA9548A_SEL_CH7)
```

I could do other way too:
```
  addr = TCA9548A_BASEADDR0 + devno;

  if (addr < TCA9548A_BASEADDR0 || addr > TCA9548A_BASEADDR7)
    {
      i2cerr("Invalid devno: %d! Valid range: 0 to 7!\n", devno);
      return -EINVAL;
    }
```

Reply:
Done! Thank you!